numpy/random/mtrand/mtrand.pyx

```diff
@@ -681,8 +681,13 @@ cdef class RandomState:
                 with self.lock:
                     rk_seed(idx, self.internal_state)
         except TypeError:
-            obj = np.asarray(seed).astype(np.int64, casting='safe')
-            if ((obj > int(2**32 - 1)) | (obj < 0)).any():
+            obj = np.asarray(seed)
+            if obj.size == 0:
+                raise ValueError("Seed must be non-empty")
+            obj = obj.astype(np.int64, casting='safe')
+            if obj.ndim != 1:
+                raise ValueError("Seed array must be 1-d")
+            if ((obj >= 2**32) | (obj < 0)).any():
```

Comment:
Sorry, just spotted that making this `2**32` makes this inconsistent with the `if` above - can you change the other one too? (or change this one back)